salt/states/htpasswd.py

```diff
@@ -67,6 +80,10 @@ def user_exists(name, password=None, htpasswd_file=None, options='',
             ret['result'] = False
             ret['comment'] = useradd_ret['stderr']
             return ret
-    ret['result'] = True
+
+    if __opts__['test']:
+        ret['result'] = None
+    else:
+        ret['result'] = True
```

Comment:
Totally trivial, but could be shortened to:

    ret['result'] = bool(__opts__['test']) or None